### DISCUSSION TITLE
[BUG] Error using bulk import via CLI

### DISCUSSION THREAD

**Discussion by thyeestes** [immich-app/immich#3360] [Q&A] [answered]
### The bug

Error asset:  {
  file: '/import/12-03-2003 034.jpg',
  reason: Error: getaddrinfo EAI_AGAIN pics.mydomain.com
      at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
    errno: -3001,
    code: 'EAI_AGAIN',
    syscall: 'getaddrinfo',
    hostname: 'pics.mydomain.com',
    config: {
      transitional: [Object],
      adapter: [Function: httpAdapter],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: Infinity,
      maxBodyLength: Infinity,
      validateStatus: [Function: validateStatus],
      headers: [Object],
      method: 'post',
      maxRedirects: 0,
      url: 'http://pics.mydomain.com:8906/api/asset/upload',
      data: [FormData]
    },
    request: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: true,
      _last: true,
      chunkedEncoding: true,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: false,
      _headerSent: true,
      _closed: true,
      socket: [Socket],
      _header: 'POST /api/asset/upload HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: multipart/form-data; boundary=--------------------------943768482232685302830456\r\n' +
        'x-api-key: 1FTNmi9eoXO0f0fYFX9sAs4AfjLLY9UpGtD7k5OJSk\r\n' +
        'User-Agent: axios/0.26.0\r\n' +
        'Host: pics.mydomain.com:8906\r\n' +
        'Connection: close\r\n' +
        'Transfer-Encoding: chunked\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/api/asset/upload',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'pics.mydomain.com',
      protocol: 'http:',
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: true,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    response: undefined,
    isAxiosError: true,
    toJSON: [Function: toJSON]
  },
  response: undefined

### The OS that Immich Server is running on

Podman

### Version of Immich Server

1.68

### Version of Immich Mobile App

1.68.0 build 91

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Not using Docker Compose
```

### Your .env content

```Shell
Not using .env all variables are declared in kube .yml file
```

### Reproduction steps

```bash
1. Browse to a folder with pictures
2. Run podman run -it --rm -v "$(pwd):/import" ghcr.io/immich-app/immich-cli:latest upload --key myapikey --server http://pics.mydomain.com:8906/api
```

### Additional information

Error does not occur on every file, only on ~10% of the files, but random. In another try the files, which had errors previously, are imported and other files have errors.

**Comment by bo0tzz** [comment, marked as answer]
> Error: getaddrinfo EAI_AGAIN pics.mydomain.com

It's having trouble either resolving the DNS name, or reaching the address that that DNS name points to (I'm not sure which exactly). This will be caused by an issue in your network or DNS configuration.

**Reply by thyeestes** [reply]
Thanks. I'll try on IP address.

**Reply by thyeestes** [reply]
IP it works. Thanks again.

Is there also an option to remove duplicates in bulk?

**Reply by bo0tzz** [reply]
Immich calculates the checksum of uploaded assets and reject any assets that are already on the server. There's currently no deduplication functionality for 'similar' assets.
